scripts/destroy.sh

```diff
@@ -123,29 +122,54 @@ for i in "${bucketList[@]}"; do
   fi
 
   # Empty the bucket
+  echo "emptying bucket: $i"
   aws s3 rm s3://$i/ --recursive
 done
 
-echo "Removing certificate from stage"
-
 restApiName=$stage-app-api
 
 restApiId=$(aws apigateway get-rest-apis | jq -r ".[] | .[] |  select(.name==\"$restApiName\") | .id|tostring")
 
+if [ "$restApiId" != "" ]; then
+  echo "Removing certificate from stage"
+
 aws apigateway update-stage \
   --rest-api-id $restApiId \
   --stage-name $stage \
   --patch-operations op=replace,path=/clientCertificateId,value="" \
   &>/dev/null
+  restApiArn="arn:aws:apigateway:us-east-1::/restapis/$restApiId/stages/$stage"
+
+  echo "Disassociating web acl from api gateway"
+
+  aws wafv2 disassociate-web-acl \
+    --resource-arn $restApiArn &>/dev/null
+
+  echo "Removed certificate from stage and disassociated web acl"
+fi
 
-echo "Removed certificate from stage"
+databaseStackName="database-$stage"
 
 # Trigger a delete for each cloudformation stack
 for i in "${stackList[@]}"; do
-  echo "Triggered stack deletion for stack: $i" 
-  aws cloudformation delete-stack --stack-name $i
+  # database stack has a bucket that the app-api stack expects to exist while deleting
+  if [ $i == $databaseStackName ]; then
+    echo "skipping database delete until app-api is finished"
+  else
+    echo "triggering stack deletion for $i"
+    aws cloudformation delete-stack --stack-name $i
+  fi
 done
 
+# if database stack exists...
+# wait for app-api to delete and then delete database stack
+if [[ "${stackList[@]}" =~ "$databaseStackName" ]]; then
+  echo "waiting for app-api stack delete to complete before deleting database stack"
+  aws cloudformation wait stack-delete-complete --stack-name "app-api-$stage"
+  echo "triggering stack deletion for $databaseStackName"
+  aws cloudformation delete-stack --stack-name $databaseStackName
+fi
```

Comment:
wait for app-api stack to delete before deleting db stack. the following error happens because the database stack gets deleted and the app-api has lambdas that rely on a bucket from that stack. If we wait, they will all delete properly rather than leaving stacks in the 'delete failed' state
![Screenshot 2023-07-31 at 2 15 44 PM](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/57802560/ae6920a3-1910-494f-b817-a943a38f5611)